crates/autopilot/src/database/onchain_order_events.rs

```diff
@@ -557,8 +558,16 @@ async fn get_quote(
         buy_amount: order_data.buy_amount,
         fee_amount: order_data.fee_amount,
         kind: order_data.kind,
-        // Original quote was made from user account, and not necessarily from owner.
-        from: order_placement.sender,
+        verification: Some(Verification {
+            // Original quote was made from user account, and not necessarily from owner.
+            from: order_placement.sender,
+            receiver: order_data.receiver.unwrap_or(order_placement.sender),
+            sell_token_source: SellTokenSource::Erc20,
+            buy_token_destination: BuyTokenDestination::Erc20,
+            // TODO do we have to get custom interactions here?
+            pre_interactions: vec![],
```

Comment:
Good question - my intuition is "no".

In theory, to correctly verify the EthFlow transaction exactly as it will execute, we should do:
```
from: owner, // i.e. the EthFLow contract
// ...
pre_interactions: vec![ethflow_pre_interaction()], // from <https://github.com/cowprotocol/services/blob/615cb7cc1ee7f45a35670948868810032882d9d2/crates/autopilot/src/database/onchain_order_events/ethflow_events.rs#L124>
```

However! the way the simulation currently works (where it wraps ETH for the `from` address automatically), it is actually not needed.

I would keep this as is for simplicity sake (and double check my assessment is not incorrect).

Reply:
When we index this event the user already sent the funds to the `EthFlow` contract, right?
In that case we actually have to do it correctly because the user might no longer have enough `WETH` for our unwrap trick during the simulation.

Reply:
Dang, you are right.

Reply:
That also begs the question how the quote flow for ethflow orders has to work.
When a user requests a quote for `ETH -> some token`. We have to make a choice whether to simulate from the user address or ethflow address.
If we simulate from the user address the quote we try to look up when indexing the ethflow event will have a different owner and we won't find it.
If we simulate from the ethflow contract we would only have to fake its `ETH` balance. That way the quote requested by the user and looked up when indexing the event should be identical so we can find it.
I guess the most reasonable approach is to drop the EthFlow verification in this PR and address that specifically in a separate PR.

Reply:
Ahh, this is particularly annoying.
Either we only have to look up the quote because it already exists since the user requested it before creating the order. Then the owner should be the user (so we compute subsidies correctly) and we need no ethflow interactions.
Or the quote doesn't exist and we create one on the fly then the owner has to be the ethflow contract to pass the simulation at that point in time.

Reply:
> If we simulate from the user address the quote we try to look up when indexing the ethflow event will have a different owner and we won't find it.

We can just "mint" ETH (i.e. specifying a `balance` in the state override for the `from` address) for the `from` address. I think this is the easiest way around this. It also means we can always simulate (W)ETH -> Token quotes which is cool.

Reply:
Had a call with Nick which concluded that we won't be requesting verified quotes when indexing onchain-orders.
The UX of that mechanism is already worse than native orders and verified quotes could make it even worse (also very complicated to get exactly right). Explained in [985658f](https://github.com/cowprotocol/services/pull/1580/commits/985658fff06e2cfca32cbe73d64b2ef9a077292d).